api/errors.go

```diff
@@ -1,38 +1,158 @@
 package api
 
+type apiErrorCode uint
+type apiErrorType string
+
 const (
 	// Public error messages (included in response objects)
 
+	// ErrParamValidationFailedCode code for param validation failed error
+	ErrParamValidationFailedCode apiErrorCode = 1
+	// ErrParamValidationFailedType type for param validation failed error
+	ErrParamValidationFailedType apiErrorType = "ErrParamValidationFailed"
+
 	// ErrDuplicatedKey error message returned when trying to insert an item with duplicated key
 	ErrDuplicatedKey = "Item already exists"
+	// ErrDuplicatedKeyCode code for duplicated key error
+	ErrDuplicatedKeyCode apiErrorCode = 2
+	// ErrDuplicatedKeyType type for duplicated key error
+	ErrDuplicatedKeyType apiErrorType = "ErrDuplicatedKey"
+
 	// ErrSQLTimeout error message returned when timeout due to SQL connection
-	ErrSQLTimeout = "The node is under heavy preasure, please try again later"
+	ErrSQLTimeout = "The node is under heavy pressure, please try again later"
+	// ErrSQLTimeoutCode code for sql timeout error
+	ErrSQLTimeoutCode apiErrorCode = 3
+	// ErrSQLTimeoutType type for sql timeout type
+	ErrSQLTimeoutType apiErrorType = "ErrSQLTimeout"
+
+	// ErrSQLNoRowsCode code for no rows error
+	ErrSQLNoRowsCode apiErrorCode = 4
```

Comment:
Here I think we need to add a text error message saying something such as: "record not found for this query and the parameters informed"